Dockerfile

```diff
@@ -19,151 +20,332 @@
 # SPDX-License-Identifier: Apache-2.0
 # License-Filename: LICENSE
 
-# Set this to the version ORT should report.
-ARG ORT_VERSION="DOCKER-SNAPSHOT"
+#------------------------------------------------------------------------
+# Use OpenJDK Eclipe Temurin Ubuntu LTS
+FROM eclipse-temurin:11-jdk-focal AS build
+
+ENV LANG=C.UTF-8
+
+# Prepare build environment to use ort scripts from here
+COPY scripts /etc/scripts
 
 # Set this to a directory containing CRT-files for custom certificates that ORT and all build tools should know about.
 ARG CRT_FILES=""
+COPY "$CRT_FILES" /tmp/certificates/
+RUN /etc/scripts/import_proxy_certs.sh \
+    && if [ -n "$CRT_FILES" ]; then \
+    /etc/scripts/import_certificates.sh /tmp/certificates/; \
+    fi
 
-# Set this to the ScanCode version to use.
-ARG SCANCODE_VERSION="30.1.0"
+#------------------------------------------------------------------------
+# Ubuntu build toolchain
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+    build-essential \
+    clang-9 \
+    clang++-9 \
+    dirmngr \
+    dpkg-dev \
+    git \
+    gnupg \
+    libbluetooth-dev \
+    libbz2-dev \
+    libc6-dev \
+    libexpat1-dev \
+    libffi-dev \
+    libgmp-dev \
+    libgdbm-dev \
+    liblzma-dev \
+    libmpdec-dev \
+    libncursesw5-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    libssl-dev \
+    make \
+    netbase \
+    openssl \
+    tk-dev \
+    tzdata \
+    uuid-dev \
+    unzip \
+    xz-utils \
+    zlib1g-dev \
+    && rm -rf /var/lib/apt/lists/*
 
-FROM eclipse-temurin:11-jdk-focal AS build
 
-COPY . /usr/local/src/ort
 
+#------------------------------------------------------------------------
+# PYTHON - Build Python as a separate component with pyenv
+FROM build as pythonbuild
+
+ARG PYTHON_VERSION=3.10.2
+ARG PYTHON2_VERSION=""
+
+ENV PYENV_ROOT=/opt/python
+RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
```

Comment:
I haven't tested it yet, but this would be a more safe way to execute an online script:

- `-e` Exit immediately if a pipeline or compound command exists with a non-zero status.
- `-u` Treat unset variables and parameters other than the special parameters "@" and "*" as an error when performing parameter expansion.
- `-x` After expanding each simple command, for command, case command, select command, or arithmetic for command, display the expanded value of PS4, followed by the command and its expanded arguments or associated word list.
- `-o pipefail` If set, the return value of a pipeline is the value of the last (rightmost) command to exit with a non-zero status, or zero if all commands in the pipeline exit successfully.`

```suggestion
RUN bash -c "set -euxo pipefail && curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash"
```

Reply:
- `-e` -> there is just one command, makes no difference
- `u` -> there are no variables, makes no difference
- `x` docker also shows you what is executed, for a single line command that might just add noise

it is also not necessary to wrap that in bash, the following also works:

```suggestion
RUN set -o pipefail && curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
```

## your options are not passed on to the `pyenv-installer` script
Also, the bash options are not applied to the execution of `pyenv-installer`, as you can see in the following example:
```
> cat test.sh
───────┬
       │ File: test.sh
───────┼
   1   │ #!/usr/bin/env bash
   2   │
   3   │ set -e
   4   │ cat <<EOF | bash
   5   │ false
   6   │
   7   │ echo "command after false"
   8   │ EOF
───────┴
> ./test.sh
command after false
```

If you want to add these you have to do the following:

```suggestion
RUN curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash -euxo pipefail
```

### But you should not apply these options to scripts that are not written with these options in mind
This changes the behaviour of the script and if it was not developed with these options in mind it will behave differently. So you should not apply these options to a random script outside of your control

And that script already has `-e` set, and would set `-x` if the right env variable is provided:
https://github.com/pyenv/pyenv-installer/blob/49fba599e872bc761858ea6f700271fb6dcb5a97/bin/pyenv-installer#L3-L4

Reply:
So, leave it all out then?

Reply:
I would suggest to do just the following change:
```suggestion
RUN set -o pipefail && curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer | bash
```